homeassistant/components/inels/config_flow.py

```diff
@@ -0,0 +1,273 @@
+            )
+
+            if test_connect is None:
+                user_input[CONF_DISCOVERY] = True
+                return self.async_create_entry(
+                    title=TITLE,
+                    data={
+                        CONF_HOST: user_input[CONF_HOST],
+                        CONF_PORT: user_input[CONF_PORT],
+                        CONF_USERNAME: user_input.get(CONF_USERNAME),
+                        CONF_PASSWORD: user_input.get(CONF_PASSWORD),
+                        MQTT_TRANSPORT: user_input.get(MQTT_TRANSPORT),
+                        CONF_DISCOVERY: True,
+                    },
+                )
+
+            errors["base"] = connect_val_to_error(test_connect)
+        else:
+            user_input = {}
+
+        fields = OrderedDict()
+        fields[vol.Required(CONF_HOST, default=user_input.get(CONF_HOST))] = str
+        fields[
+            vol.Required(
+                CONF_PORT,
+                default=1883
+                if user_input.get(CONF_PORT) is None
+                else user_input.get(CONF_PORT),
+            )
+        ] = vol.Coerce(int)
+        fields[vol.Optional(CONF_USERNAME, default=user_input.get(CONF_USERNAME))] = str
+        fields[vol.Optional(CONF_PASSWORD, default=user_input.get(CONF_PASSWORD))] = str
+        fields[vol.Required(MQTT_TRANSPORT, default="tcp")] = vol.In(
+            ["tcp", "websockets"]
+        )
+
+        return self.async_show_form(
+            step_id="setup", data_schema=vol.Schema(fields), errors=errors
+        )
+
+    async def async_step_hassio(self, discovery_info: HassioServiceInfo) -> FlowResult:
+        """Receive a Hass.io discovery."""
+        await self._async_handle_discovery_without_unique_id()
+        self._hassio_discovery = discovery_info.config
+
+        return await self.async_step_confirm()
+
+    async def async_step_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Confirm a discovery."""
+        errors = {}
+        assert self._hassio_discovery
+
+        if user_input is not None:
+            data = self._hassio_discovery
+            test_connect = await self.hass.async_add_executor_job(
+                try_connection,
+                self.hass,
+                data[CONF_HOST],
+                data[CONF_PORT],
+                data.get(CONF_USERNAME),
+                data.get(CONF_PASSWORD),
+                data.get(MQTT_TRANSPORT),
+            )
+
+            if test_connect is None:
+                return self.async_create_entry(
+                    title=TITLE,
+                    data={
+                        CONF_HOST: data[CONF_HOST],
+                        CONF_PORT: data[CONF_PORT],
+                        CONF_USERNAME: data.get(CONF_USERNAME),
+                        CONF_PASSWORD: data.get(CONF_PASSWORD),
+                        MQTT_TRANSPORT: data.get(MQTT_TRANSPORT),
+                        CONF_DISCOVERY: True,
```

Comment:
What's up with CONF_DISCOVERY? You always set it to True and I'm not sure if you even use it?

Reply:
I'll be quite honest with you, I wasn't the one who wrote this part of the code but the setup gets stuck if I do remove it, so I do use it somewhere. The integration does have a discovery step through MQTT, so it kind of makes sense?